sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +374,199 @@ def transform(self, X):
             X[coordinates] = values
 
         return X
+
+
+class MissingIndicator(BaseEstimator, TransformerMixin):
+    """Binary indicators for missing values.
+
+    Parameters
+    ----------
+    missing_values : integer or np.NaN, optional (default=np.NaN)
+        The placeholder for the missing values. All occurrences of
+        ``missing_values`` will be represented as ones.
+
+    features : {'auto' (default), 'all', array-like of int}
+        If "all", mask will represent all features.
+        If "auto", mask will only represent features with missing values
+        during fit time.
+        If mask/indices, mask will only represent features in the
+        indices or mask.
+
+    sparse : boolean or "auto", optional (default="auto")
+        If True, the transformed ``X`` will be a sparse matrix.
+        If False, the transformed ``X`` will be a numpy array.
+        If "auto", the transformed ``X`` will be of same type as input.
+
+    error_on_new : boolean, optional (default=True)
+        If True, transform will raise an error when there are features with
+        missing values in transform but have no missing values in fit
+        This is applicable only when ``features`` = "auto"
+
+    Example
+    -------
+    >>> from sklearn.preprocessing import MissingIndicator
+    >>> import numpy as np
+    >>> X1 = np.array([
+    ...       ["NaN",  1,  3],
```

Comment:
For me by using a string NaN, this results in a full string array. So I think this should be np.nan

Reply:
Please address this.